### PR TITLE
[csi/aws] Bump templates + add support for warm pools

### PR DESCRIPTION
**Pull request by dntosas**
AWS EBS CSI Driver:
    - Update manifests
    - Bump components version
    - Add API capability of setting Version + VolumeLimit
    - Add support for warm pools
    - Remove snapshot-controller resources as it should be independent from
    any CSI driver, check here --> https://kubernetes.io/blog/2020/12/10/kubernetes-1.20-volume-snapshot-moves-to-ga/#how-to-deploy-volume-snapshots
 
Signed-off-by: dntosas <ntosas@gmail.com>

### CODE REVIEW

**Comment by dntosas** [comment]
qq: where IAM resources that described in [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/example-iam-policy.json) are handled?

**Comment by olemarkus** [comment]
they are handled in `pkg/model/iam/iam_builder.go`.

It would be nice to collect all the privs needed for CSI to one function though, to make this more maintainable. That would also allow us to use IRSA for this addon further down the road.

**Comment by dntosas** [comment]
@olemarkus are not IAM permissions for attaching etc volumes should be present on regular nodes role also? without them how will the pods of node daemonSet can be allowed to do such operations?

**Comment by olemarkus** [comment]
As I understand it, it is the ebs-csi-controller that does the AWS calls. The purpose of the DaemonSet is to extend kubelet with the driver specific functionality.

**Comment by dntosas** [comment]
> As I understand it, it is the ebs-csi-controller that does the AWS calls. The purpose of the DaemonSet is to extend kubelet with the driver specific functionality.

y makes sense ^^

**Comment by johngmyers** [comment]
I don't see the code for calculating the default value of `Version`

**Comment by dntosas** [comment]
@johngmyers i thought something like [this](https://github.com/kubernetes/kops/blob/e8c8b92b95b49aba161f13a67f080b28c39abe97/upup/models/cloudup/resources/addons/aws-ebs-csi-driver.addons.k8s.io/k8s-1.17.yaml.template#L249) should be enough ^^ if there is a more proper way or place to implement this logic, can you give a hint on where to look?

**Comment by johngmyers** [comment]
@dntosas Okay, that does work; I had missed that. Unfortunately it is error prone to have the version in so many places. If one gets missed on an update that is unlikely to be caught in code review.

The other way is to define a template function, like https://github.com/kubernetes/kops/blob/dbd23473efe25c7eca1576f462b60233cd715d4d/upup/pkg/fi/cloudup/template_functions.go#L588-L608

**Comment by dntosas** [comment]
@olemarkus @johngmyers thanks for your help :)

pushed changes, i think we are better now ^^

File: pkg/apis/kops/componentconfig.go [hunk 1]
```diff
@@ -838,6 +838,16 @@ type CloudConfiguration struct {
 type AWSEBSCSIDriver struct {
 	//Enabled enables the AWS EBS CSI driver
 	Enabled *bool `json:"enabled,omitempty"`
+
+	// Version of the CSI controller container
```
Comment (olemarkus) [inline pkg/apis/kops/componentconfig.go:842]: Nit: go comments should have proper punctuation.

File: pkg/model/components/awsebscsidriver.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+var _ loader.OptionsBuilder = &AWSEBSCSIDriverOptionsBuilder{}
+
+func (b *AWSEBSCSIDriverOptionsBuilder) BuildOptions(o interface{}) error {
+	clusterSpec := o.(*kops.ClusterSpec)
+	c := clusterSpec.CloudConfig.AWSEBSCSIDriver
+	if c == nil {
+		return nil
+	}
+
+	if c.Enabled == nil {
```
Comment (olemarkus) [inline pkg/model/components/awsebscsidriver.go:39]: nit: this check is typically noops as we tend to use `if fi.BoolValue()` which defaults to false.
And you _have_ to use `{{ if WithDefaultBool <defaultVaule> }}` in templates as templates will only check if the pointer is nil or not.

**Review by olemarkus — CHANGES_REQUESTED** [review]

**Comment by olemarkus** [comment]
Just a few nits, otherwise looks good to go to me.

**Review by olemarkus — APPROVED** [review]

**Pull request merged**